CONTRIBUTING.md

```diff
@@ -48,7 +48,7 @@ Thank you for your interest in contributing to Azure SDK for C.
 
 Project contains files to work on Windows, Mac or Linux based OS.
 
-### VCPKG
+### Windows
 
 vcpkg is the easiest way to have dependencies installed. It downloads packages sources, headers and build libraries for whatever TRIPLET is set up (platform/arq).
```

Comment:
We now seem to be repeating this paragraph twice. Can we avoid that by restructuring things?

Reply:
I'm repeating the paragraph because:
1. It's not currently clear how to get this running on Linux (suggestions are not complete)
2. We should always keep Windows vs Linux vs Mac separate as much as possible. I don't have Mac separated completely because it doesn't work today at all.

Reply:
I agree with the restructuring - let's keep that in a separate PR. I need to add some things to VCPkg for Paho samples.
Based on our discussion, we need to have:
1. Something for people wanting to _use_ the SDK - how to use VCPKG or other things to build and run samples.
2. Something for people wanting to contribute (this document). For these, _maybe_ we need to split into devs for a particular client since we have very different requirements.